### PR TITLE
feat: additional parameters for anf

### PR DESCRIPTION
**Pull request by bobmclane999**
## Description
Additional parameters for NetApp - 

- Active Directory encryption, Kerberos, ldapoverTLS
- Volumes - Cool Access, Zones, replication, 
- Backup Policies, Vaults & backups

## Pipeline Reference

| Pipeline |
| -------- |
|  [avm.res.net-app.net-app-account](https://github.com/bobmclane999/bicep-registry-modules/actions/workflows/avm.res.net-app.net-app-account.yml)        |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by bobmclane999** [comment]
@jtracey93 Please do you mind give this PR the once over ? As only 2nd one :)
@fbinotto Please can you review as soon as you can so we can crack on with our File Share Migration ANF deployments :)

Thanks guys!

**Comment by jtracey93** [comment]
hey @bobmclane999, i think you ned to pull in and merge and resolve those merge conflicts. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

Something like:

```
git fetch upstream
git merge upstream main
<RESOLVE CONFLICTS>
git commit
git push

<RE RUN SET-AVMMODULE SCRIPT>
git commit
git push
```

**Comment by bobmclane999** [comment]
@jtracey93  - erm.... think I did something wrong .. I fixed the conflicts in github removing the <<< === >>> committed (as struggling with the commands) then done a sync in vscode, reran the AVVMModule and now lots of red... :'-(
image
Any ideas on how to fix? Sorry I know you're a busy man and the football is on soon so maybe one for tomorrow ...

**Comment by fbinotto** [comment]
Take your time @bobmclane999 and @jtracey93 

I still have to merge https://github.com/Azure/bicep-registry-modules/pull/2636 before I consider this PR. And that PR has some of the changes proposed here.

**Comment by bobmclane999** [comment]
Hi @fbinotto yeah we had the issues with the resourceId duplication on the WAF aligned tests similar to #2636 so had to make a similar change to get the test working - 
image

**Comment by fbinotto** [comment]
> Hi @fbinotto yeah we had the issues with the resourceId duplication on the WAF aligned tests similar to #2636 so had to make a similar change to get the test working - image
> 
> Sorry to push :( but do you have a timescale on approving the other PR (well and this one :)) as we are mid project for a File Migration using ANF so deadlines are a little tight hence all the changes... tyvm Shaun

It should be merged today or tomorrow, then you should be able to update your PR and I can approve.

**Comment by jtracey93** [comment]
Hey @bobmclane999, I see @fbinotto's PR is now merged thanks to @AlexanderSehr's approval 👍 

You can now pull in these changes into your branch and then re-run the test and README generation etc. and then @fbinotto should be good to review and get this one approved and merged too 👍

**Comment by fbinotto** [comment]
@bobmclane999 please do so before tomorrow night as I will be on leave for a week.

**Comment by jtracey93** [comment]
@fbinotto are you happy for me to approve in your absence if needed?

**Comment by jtracey93** [comment]
@fbinotto looks like its good for a review, lets see if we can get it in before you go, if not let me know

File: avm/res/net-app/net-app-account/capacity-pool/main.bicep [hunk 1]
```diff
@@ -91,14 +94,36 @@ module capacityPool_volumes 'volume/main.bicep' = [
...
+      networkFeatures: contains(volume, 'networkFeatures') ? volume.networkFeatures : networkFeatures
       creationToken: contains(volume, 'creationToken') ? volume.creationToken : volume.name
       usageThreshold: volume.usageThreshold
       protocolTypes: contains(volume, 'protocolTypes') ? volume.protocolTypes : []
       subnetResourceId: volume.subnetResourceId
       exportPolicyRules: contains(volume, 'exportPolicyRules') ? volume.exportPolicyRules : []
+      zones: contains(volume, 'zones') ? volume.zones : []
       roleAssignments: contains(volume, 'roleAssignments') ? volume.roleAssignments : []
-      networkFeatures: volume.?networkFeatures
-      zones: volume.?zones
```
Comment (fbinotto) [inline avm/res/net-app/net-app-account/capacity-pool/main.bicep:101]: These should not be removed. This syntax automatically pull the default from the referenced volume module.
Remove your lines 97 and 103 and leave these.

**Review by fbinotto — DISMISSED** [review]
LGTM. Just one minor change and we can merge,

**Comment by fbinotto** [comment]
> @fbinotto are you happy for me to approve in your absence if needed?

Yes, go ahead. I have requested only a minor change.

**Comment by bobmclane999** [comment]
@fbinotto / @jtracey93  - have put networkFeatures and Zones params back and previous and submitted the PR again for review... thanks muchly Shaun :)
Have a great week off @fbinotto !

**Comment by jtracey93** [comment]
@bobmclane999 did you re-run the `Set-AVMModule` script post making those changes and re run the tests?

**Comment by bobmclane999** [comment]
> @bobmclane999 did you re-run the `Set-AVMModule` script post making those changes and re run the tests?

Hi @jtracey93  actually no I didn't let me do and let you know when I've re-submitted tyvm

**Comment by bobmclane999** [comment]
Hi @jtracey93  - re-ran a test after another commit (post Set-AVMModule) and have another badge here - 

[avm.res.net-app.net-app-account](https://github.com/bobmclane999/bicep-registry-modules/actions/workflows/avm.res.net-app.net-app-account.yml)

Is this the correct place to put it ?

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**